beam_jobs/base_model_validator_test.py

```diff
@@ -0,0 +1,195 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+import unittest
+
+import apache_beam as beam
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that, equal_to
+from apache_beam.runners.direct.direct_runner import DirectRunner
+
+from beam_jobs import base_model_validator
+from core.platform import models
+
+(base_models, user_models) = models.Registry.import_models(
+    [models.NAMES.base_model, models.NAMES.user])
+
+
+class MockModel(base_models.BaseModel):
+    pass
+
+
+class BaseModelValidatorTests(unittest.TestCase):
+
+    def setUp(self):
+        self.now = datetime.datetime.utcnow()
+        self.year_ago = datetime.datetime.utcnow() - datetime.timedelta(weeks=52)
+        self.year_later = datetime.datetime.utcnow() + datetime.timedelta(weeks=52)
+
+    def test_validate_deleted_reports_error_for_old_deleted_model(self):
+
+        with TestPipeline(runner=DirectRunner()) as p:
+            pcoll = p | beam.Create([
+                MockModel(
+                    id='123',
+                    deleted=True,
+                    created_on=self.year_ago,
+                    last_updated=self.year_ago
+                )
+            ])
+
+            errors = (pcoll | beam.ParDo(
+                base_model_validator.ValidateDeleted())
+                .with_outputs())
+
+            assert_that(
```

Comment:
...and make calls as:
```suggestion
beam_testing_utils.assert_that(
```